tests/test_other.py

```diff
@@ -7596,6 +7596,57 @@ def test_binaryen_ignore_implicit_traps(self):
       print('sizes:', sizes)
       assert sizes[1] < sizes[0], 'ignoring implicit traps must reduce code size'
 
+  def test_binaryen_methods(self):
+    for method_init in ['interpret-asm2wasm', 'interpret-s-expr', 'asmjs', 'interpret-binary', 'asmjs,interpret-binary', 'interpret-binary,asmjs']:
+      # check success and failure for simple modes, only success for combined/fallback ones
+      for success in [1, 0] if ',' not in method_init else [1]:
+        method = method_init
+        command = [PYTHON, EMCC, '-o', 'a.wasm.js', '-s', 'BINARYEN=1', path_from_root('tests', 'hello_world.c')]
+        command += ['-s', 'BINARYEN_METHOD="' + method + '"']
+        print([method, ' : ', ' '.join(command), ' => ', success])
+        subprocess.check_call(command)
+
+        see_polyfill =  'var WasmJS = ' in open('a.wasm.js').read()
+
+        if method and 'interpret' not in method:
+          assert not see_polyfill, 'verify polyfill was not added - we specified a method, and it does not need it'
+        else:
+          assert see_polyfill, 'we need the polyfill'
+
+        def break_cashew():
+          asm = open('a.wasm.asm.js').read()
+          asm = asm.replace('"almost asm"', '"use asm"; var not_in_asm = [].length + (true || { x: 5 }.x);')
+          asm = asm.replace("'almost asm'", '"use asm"; var not_in_asm = [].length + (true || { x: 5 }.x);')
+          with open('a.wasm.asm.js', 'w') as o: o.write(asm)
```

Comment:
Is this going to work on all OSes (i.e. you open the file for reading, and then open it for writing without closing it)?

Reply:
Good point, fixed.